python/pyarrow/tests/test_io.py

```diff
@@ -1137,7 +1142,14 @@ def test_memory_zero_length(tmpdir):
 
 
 def test_memory_map_large_seeks():
-    check_large_seeks(pa.memory_map)
+    if sys.maxsize >= 2**32:
+        expected_error = None
```

Comment:
I think you can simplify the other function by passing `contextlib.nullcontext()` here.

Reply:
Ah, good point.

Reply:
Well, I think you could do something like:

```python
with expected_error:
    f = file_factory(filename)
with f:
    ...
```

I hope I'm not being too persistent.

Reply:
In this case, `expected_error` would swallow the error and `with f` would error out because `f` would be undefined :-)